### PR TITLE
fix(flows): correct deduplication

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-2186/broken-flow-deduplication

- Fix deduplication 
- Add integration tests since it's probably the third or fourth time
- Add missing types

### CODE REVIEW

File: packages/server/lib/controllers/v1/integrations/providerConfigKey/flows/getFlows.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export const getIntegrationFlows = asyncWrapper<GetIntegrationFlows>(async (req,
 function containsSameEndpoint(flowA: NangoSyncConfig, flowB: NangoSyncConfig) {
     for (const endpointObjA of flowA.endpoints) {
         for (const endpointObjB of flowB.endpoints) {
-            if (endpointObjB.method && endpointObjA.method && endpointObjB.path === endpointObjA.path) {
+            if (endpointObjB.method === endpointObjA.method && endpointObjB.path === endpointObjA.path) {
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/v1/integrations/providerConfigKey/flows/getFlows.ts:63]: this is the fix 🤦🏻

File: packages/jobs/lib/execution/sync.integration.test.ts [hunk 2]
```diff
@@ -326,22 +321,22 @@ async function populateRecords(
 }
 
 async function seeds(records: UnencryptedRecordData[], trackDeletes: boolean) {
-    const envName = Math.random().toString(36).substring(7);
-    const env = await seeders.createEnvironmentSeed(0, envName);
-    const providerConfigKey = Math.random().toString(36).substring(7);
-    const model = Math.random().toString(36).substring(7);
+    const { env } = await seeders.seedAccountEnvAndUser();
+    const model = 'GithubIssue';
```
Comment (bodinsamuel) [inline packages/jobs/lib/execution/sync.integration.test.ts:325]: Changed this test to be non-random (except the account id), bc there was an issue after I changed the seeders

File: packages/server/lib/controllers/v1/integrations/providerConfigKey/flows/getFlows.ts (hunk 1 above)
Comment (TBonnin) [reply packages/server/lib/controllers/v1/integrations/providerConfigKey/flows/getFlows.ts:63]: 🙈

File: packages/shared/lib/seeders/config.seeder.ts [hunk 3]
```diff
@@ -25,10 +25,14 @@ export const createConfigSeeds = async (env: DBEnvironment): Promise<void> => {
...
-export const createConfigSeed = async (env: DBEnvironment, unique_key: string, provider: string): Promise<ProviderConfig | null> => {
-    return configService.createProviderConfig({
+export const createConfigSeed = async (env: DBEnvironment, unique_key: string, provider: string): Promise<ProviderConfig> => {
+    const created = await configService.createProviderConfig({
         unique_key,
         provider,
         environment_id: env.id
     } as ProviderConfig);
+    if (!created) {
+        throw new Error('failed to created to provider config');
```
Comment (TBonnin) [inline packages/shared/lib/seeders/config.seeder.ts:35]:
```suggestion
        throw new Error('failed creating provider config');
```

**Review by TBonnin — APPROVED** [review]

**Pull request merged**